packages/cli/src/WorkflowRunner.ts

```diff
@@ -54,12 +56,48 @@ export class WorkflowRunner {
 		private readonly workflowStaticDataService: WorkflowStaticDataService,
 		private readonly nodeTypes: NodeTypes,
 		private readonly permissionChecker: PermissionChecker,
+		private readonly ownershipService: OwnershipService,
 	) {
 		if (this.executionsMode === 'queue') {
 			this.jobQueue = Container.get(Queue);
 		}
 	}
 
+	async init() {
+		if (this.executionsMode === 'queue') return;
+
+		const executionIds = await this.executionRepository.getNewExecutionIds();
```

Comment:
We're fetching here and again below with `findSingleExecution`. Can we do a single query for the full unflattened data for all new executions?